src/PlusImageProcessing/vtkPlusTransverseProcessEnhancer.cxx

```diff
@@ -339,47 +349,118 @@ PlusStatus vtkPlusTransverseProcessEnhancer::WriteConfiguration(vtkXMLDataElemen
 
-  XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(imageProcessingOperations, processingElement, "ImageProcessingOperations");
-  if (this->GaussianEnabled)
-  {
-    XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(gaussianParameters, processingElement, "GaussianSmoothing");
-    gaussianParameters->SetDoubleAttribute("GaussianStdDev", this->GaussianStdDev);
-    gaussianParameters->SetDoubleAttribute("GaussianKernelSize", this->GaussianKernelSize);
-  }
-
-  if (this->ThresholdingEnabled)
-  {
-    XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(thresholdingParameters, processingElement, "Thresholding");
-    thresholdingParameters->SetDoubleAttribute("ThresholdInValue", ThresholdInValue);
-    thresholdingParameters->SetDoubleAttribute("ThresholdOutValue", ThresholdOutValue);
-    thresholdingParameters->SetDoubleAttribute("LowerThreshold", LowerThreshold);
-    thresholdingParameters->SetDoubleAttribute("UpperThreshold", UpperThreshold);
-  }
-
-  if (this->EdgeDetectorEnabled)
-  {
-    processingElement->SetAttribute("EdgeDetectorEnabled", "TRUE");
-  }
-
-  if (this->IslandRemovalEnabled)
-  {
-    XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(islandRemovalParameters, processingElement, "IslandRemoval");
-    islandRemovalParameters->SetIntAttribute("IslandAreaThreshold", IslandAreaThreshold);
-  }
-
-  if (this->ErosionEnabled)
-  {
-    XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(erosionParameters, processingElement, "Erosion");
-    erosionParameters->SetVectorAttribute("ErosionKernelSize", 2, this->ErosionKernelSize);
-  }
-
-  if (this->DilationEnabled)
-  {
-    XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(dilationParameters, processingElement, "Dilation");
-    dilationParameters->SetVectorAttribute("DilationKernelSize", 2, this->DilationKernelSize);
-  }
-
-  return PLUS_SUCCESS;
+  processingElement->SetIntAttribute("NumberOfScanLines", NumberOfScanLines);
+  processingElement->SetIntAttribute("NumberOfSamplesPerScanLine", NumberOfSamplesPerScanLine);
+
+  XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(scanConversion, processingElement, "ScanConversion");
+  scanConversion->SetAttribute("TransducerName", "Ultrasonix_C5-2");
+  scanConversion->SetAttribute("TransducerGeometry", "CURVILINEAR");
+  scanConversion->SetDoubleAttribute("RadiusStartMm", this->RadiusStartMm);
+  scanConversion->SetDoubleAttribute("RadiusStopMm", this->RadiusStopMm);
+  scanConversion->SetIntAttribute("ThetaStartDeg", this->ThetaStartDeg);
+  scanConversion->SetIntAttribute("ThetaStopDeg", this->ThetaStopDeg);
+  scanConversion->SetVectorAttribute("OutputImageSizePixel", 2, this->OutputImageSizePixel);
+  scanConversion->SetVectorAttribute("TransducerCenterPixel", 2, this->TransducerCenterPixel);
+  scanConversion->SetVectorAttribute("OutputImageSpacingMmPerPixel", 2, this->OutputImageSpacingMmPerPixel);
+
+	XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(imageProcessingOperations, processingElement, "ImageProcessingOperations");
+
+	if (this->ConvertToLinesImage)

```

Comment:
Use XML_WRITE_BOOL_ATTRIBUTE macro. Use XML_WRITE macros to simplify writing of other attributes as well.